React/Modules/RCTUIManager.m

```diff
@@ -224,6 +224,8 @@ @implementation RCTUIManager
   NSDictionary *_componentDataByName;
 
   NSMutableSet<id<RCTComponent>> *_bridgeTransactionListeners;
+  NSMutableSet<id<RCTUIManagerObserver>> *_uiManagerObservers;
```

Comment:
Native modules never seemed to get deallocated during runtime so it shouldn't be an issue, even if they did shouldn't leak since we are removing the observer.